org-code-javabuilder/lib/src/main/java/org/code/javabuilder/ProjectFileLoader.java

```diff
@@ -0,0 +1,66 @@
+package org.code.javabuilder;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.Callable;
+
+/**
+ * Fetches an individual file from a student's project. This should be used with the
+ * {dashboardHost}/v3/files/{channelId}/{fileName} API. It has no permissions to use AWS S3. This
+ * should be used asynchronously so multiple files can be retrieved at once.
+ */
+public class ProjectFileLoader implements Callable<Boolean> {
+  private final ProjectFile projectFile;
+  private final String fileUrl;
+
+  /**
+   * @param projectFile The in-memory representation of the given file
+   * @param fileUrl The URL to use to fetch the file
+   */
+  public ProjectFileLoader(ProjectFile projectFile, String fileUrl) {
+    this.projectFile = projectFile;
+    this.fileUrl = fileUrl;
+  }
+
+  /**
+   * @return true if the load succeeds
+   * @throws UserFacingException If any errors occur while fetching the file.
+   */
+  @Override
+  public Boolean call() throws UserFacingException {
+    try {
+      URL url = new URL(fileUrl);
+      HttpURLConnection con = (HttpURLConnection) url.openConnection();
```

Comment:
How are we ensuring that a student can only access their own files?  (I was expecting to see us pass a security token of some sort here?)

Reply:
We'll need to do that for the `PUT` when we implement it later, but for now, we're just using the public `GET` endpoint.

Reply:
Oh, is it by design that we allow anyone to retrieve any student's code?  That seems...  unexpected?

Reply:
Hmm. Double checking with the infra team.

Reply:
Sounds like it's by design. I believe it's a requirement of our project sharing feature (i.e. share between students, share to phone, etc.). More details here: https://codedotorg.slack.com/archives/C03CK49G9/p1617841767304900